target/linux/rockchip/Makefile

```diff
@@ -0,0 +1,25 @@
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+
+ARCH:=arm
+BOARD:=rockchip
+BOARDNAME:=Rockchip RK33xx
+FEATURES:=ext4 audio usb usbgadget display gpio fpu rootfs-part boot-part
+SUBTARGETS:=rk3399
```

Comment:
Why creating subtarget when there is only single device in this target? Do You have upcoming additions of devices in queue?

Reply:
RK3328 for instance?

Reply:
Do You have it in Your queue? Also dropping the `CPU_TYPE` would allow to support rk3328 under the same target and this device is not space constrained to justify fragmentation. It works well for layerscape.

Reply:
I have a "working" rk3328, the ethernet is a problem atm

Reply:
Good, still that doesn't justify fragmentation, with current state of this PR You could use rk3399 subtaget and simply add necessary drivers and be done. But that's not rk3288 which would justify **subtargets**. For now **single** target is enough.

Reply:
Yes, we're planning on adding more hardware in the near future. I've spoken to Pine64 (we're also part of the Manjaro project) and they're sending some more Rockchip hardware my way. 
One of these platforms needs it's own subtarget and has space constraints (because the image needs to fit on an SPI flash).

Reply:
@tmn505 Currently every target present in OpenWrt has at least one subtarget (e.g. `ipq806x-generic` or `ipq40xx-generic`).

So before we are doing stuff like `rockchip-generic` I'm in favor of the current state.

Reply:
@Manawyrm that's nice, it's good to see vendors willing to give hardware for open contributions.
That is why I also suggested tuning the kernel configuration. Also if need will arise, You can move drivers not needed to boot to kernel modules which will occupy rootfs or even extroot. If U-Boot is not constrained in regard of kernel size, kernel can be as big as 15MB if the flash size is 16MB. Creating additional subtarget should be last option.

@blocktrron Ok, then the name should change. With current state, this subtarget would also support `rk3328` but the name suggests only `rk3399` which is misleading. Going further, if `CPU_TYPE` would be removed, in the same subtarget we could also add `rk1808` devices, then again the name would be misleading. We could change that as the new devices would be added but this would leave dangling directories for targets no longer updated, occupying storage on download server. It's best to chose right name at the beginning. I would opt for `armv8` or `64` or both combined, it the need would arise then something like `armv8_tiny` could be created.

Reply:
@tmn505 has a good point, looking around, what about `cortexa53` subtarget as seen in sunxi, mvebu already? Are those bad examples?

Reply:
If this subtarget could support rk1808 as well, which is `cortex-a35`, then the proposed `cortexa53` doesnt make much sense anymore, so probably the remaining ones as already suggested `64` or `armv8_64`.

Reply:
I went with cortexa53. I've spoken to @TobleMiner and some Pine64 folks and there's going to be some future additional hardware support that will require different subtargets.

Reply:
@Manawyrm That mine initial idea with `cortex-a53` wasn't that good as @tmn505 explained, there is RK1808 which could be supported with this subtarget as well as it is dual-core ARM `cortex-a35`, so my [previous comment](https://github.com/openwrt/openwrt/pull/2917#discussion_r410300701) still stands.

There seems to be [TB-96AIoT](https://www.seeedstudio.com/BeiQi-RK1808-AIoT-96Boards-Compute-SoM-p-4074.html) already available from Seeed and [SoEdge-RK1808 SoM](https://www.cnx-software.com/2019/08/05/pine64-soedge-rk1808-module-delivers-3-0-tops-via-rockchip-rk1808-soc) from pine64.

> there's going to be some future additional hardware support that will require different subtargets

Ok, then you need to provide more details then just someone told me so.

Reply:
@ynezz Hm, ok. Let's just make it generic then. 
How do we want to name the subtarget? 

armv8_64b ? (identical to layerscape?)

Reply:
Naming is hard :) I would just go with `armv8`, it's probably unlikely, that Rockchip is going to use some older 32-bit ARMv8 core.

Reply:
`armv8` sounds good to be, too